.github/workflows/build.yaml

```diff
@@ -1,8 +1,98 @@
 name: Build
 on:
+  schedule:
+    # Run nightly at 00:00
+    - cron: '0 0 * * *'
+
   workflow_dispatch:
+    inputs:
+      otp-ref-name:
+        required: true
+      otp-ref:
+        required: true
+      openssl-version:
+        default: "3.1.6"
+      wxwidgets-version:
+        default: "3.2.6"
 jobs:
-  build:
+  schedule:
+    name: "Schedule nightly"
+    if: github.event_name == 'schedule'
     runs-on: ubuntu-latest
+    permissions:
+      actions: write
     steps:
-      - run: echo "Hello, World!"
+      - name: "Schedule nightly"
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          ref=$(gh api repos/erlang/otp/commits/master --jq .sha)
+          gh workflow run --repo "${GITHUB_REPOSITORY}" build.yml --field otp-ref-name=master --field otp-ref=$ref
+
+          ref=$(gh api repos/erlang/otp/commits/maint --jq .sha)
+          gh workflow run --repo "${GITHUB_REPOSITORY}" build.yml --field otp-ref-name=maint --field otp-ref=$ref
+
+  build_macos:
+    name: "Build ${{ inputs.otp-ref-name }}-macos-${{ matrix.pair.arch }}"
+    if: github.event_name == 'workflow_dispatch'
+    runs-on: ${{ matrix.pair.os }}
+    permissions:
+      actions: write
+      contents: write
+    env:
+      OTP_REF_NAME: ${{ inputs.otp-ref-name }}
+      OPENSSL_VERSION: ${{ inputs.openssl-version }}
+      OPENSSL_DIR: /tmp/builds/openssl-${{ inputs.openssl-version }}-macos-${{ matrix.pair.arch }}
+      WXWIDGETS_VERSION: ${{ inputs.wxwidgets-version }}
+      WXWIDGETS_DIR: /tmp/builds/wxwidgets-${{ inputs.wxwidgets-version }}-macos-${{ matrix.pair.arch }}
+      OTP_DIR: /tmp/builds/otp-${{ inputs.otp-ref-name }}-${{ inputs.otp-ref }}-openssl-${{ inputs.openssl-version }}-wxwidgets-${{ inputs.wxwidgets-version }}-macos-${{ matrix.pair.arch }}
+      OTP_REF: ${{ inputs.otp-ref }}
+    strategy:
+      matrix:
```

Comment:
Would it make sense to add:

* `fail-fast: true` - e.g. failing 13 shouldn't prevent 14 from getting released, or vice-versa
* `max-parallel: 1` - this way we're bound to have less concurrency issues with GitHub runners (I've had instances where the filesystem seems shared - or some kind of caching is taking place - that this option eased)

Reply:
I added fail-fast: false for now, thanks!